installers/MacInstaller/make_package.rb

```diff
@@ -0,0 +1,587 @@
+def stop_if_bundle_not_self_contained(basepath, extra = [])
+  search_patterns = ["#{basepath}/**/*.dylib",
+                     "#{basepath}/**/*.so"]
+  search_patterns += extra
+  problem_libraries = []
+  search_patterns.each do |pattern|
+    Dir[pattern].each do |library|
+      find_dependencies(Pathname.new(library)).each do |dependency|
+        if dependency.include?(HOMEBREW_PREFIX)
+          problem_libraries << library
+          break
+        end
+      end
+    end
+  end
+
+  unless problem_libraries.empty?
+    msg = 'The bundle is not self contained. The following '\
+          "libraries link to others outside of the bundle:\n"\
+          "#{problem_libraries.join('\n')}"
+    fatal(msg)
+  end
+end
+
+#---------------------------------------------------------
+# Main script
+#---------------------------------------------------------
+if (ARGV.length < 2) || (ARGV.length > 3)
+  puts 'Usage: make_package bundle-path python-exe [paraview-build-dir]'
+  puts '  - bundle-path: Path of bundle to fix'
+  puts '  - python-exe: Path to Python executable to bundle. The whole Python.framework is bundled.'
+  puts '  - paraview-build-dir: Optional path to ParaView build dir'
+  exit 1
+end
+
+# Host paths
+host_python_exe = Pathname.new(ARGV[1])
+fatal("Python executable #{python_exe} not found") unless host_python_exe.exist?
+$PARAVIEW_BUILD_DIR = ARGV[2] if ARGV.length == 3
+if $PARAVIEW_BUILD_DIR.length == 0
+  # assume a non vates build
+  # set to some string that will not be found at the start of an otool dependency
+  $PARAVIEW_BUILD_DIR = '__NOTUSED__'
+end
+
+# Bundle paths
+bundle_path = Pathname.new(ARGV[0])
+contents_macos = bundle_path + 'Contents/MacOS'
+contents_frameworks = bundle_path + 'Contents/Frameworks'
+# additional executables not detectable by dependency analysis
+executables = ["#{contents_macos}/MantidNexusParallelLoader"]
+
+# check we have a known bundle
+if bundle_path.to_s.end_with?('MantidWorkbench.app')
+  bundled_packages = BUNDLED_PY_MODULES + BUNDLED_PY_MODULES_WORKBENCH
+  bundled_qt_plugins = QT_PLUGINS_COMMON + ['platforms', 'styles']
+  host_qt_plugins_dir = QT5_PLUGINS_DIR
+elsif bundle_path.to_s.end_with?('MantidPlot.app')
+  bundled_packages = BUNDLED_PY_MODULES + BUNDLED_PY_MODULES_MANTIDPLOT
+  bundled_qt_plugins = QT_PLUGINS_COMMON
+  host_qt_plugins_dir = QT4_PLUGINS_DIR
+  executables << "#{contents_macos}/MantidPlot"
+else
+  fatal("Unknown bundle type #{bundle_path}. Expected MantidPlot.app or MantidWorkbench.app.")
+end
+
+# We start with the assumption CMake has installed all required target libraries/executables
+# into the bundle and the main layout exists.
+deploy_python_framework(contents_frameworks, host_python_exe,
+                        bundled_packages)
+install_qt_plugins(bundle_path, bundled_qt_plugins, host_qt_plugins_dir,
+                   QT_PLUGINS_BLACKLIST)
+# We choose not to use macdeployqt as it uses @executable_path so we have to essentially
```

Comment:
Agreed. `macdeployqt` never worked that well for us.

Part of me wants to go way back to when cmake/cpack created the .app bundle, but definitely not worth the time to rewrite what is already in this script.